frontend/lib/api/form/pollingstation/pollingStationUtils.test.ts

```diff
@@ -0,0 +1,417 @@
+        fields: ["data.votes_counts.votes_candidates_counts", "data.political_group_votes"],
+      }),
+    ).toBe(true);
+
+    expect(
+      isGlobalValidationResult({
+        code: "F303",
+        fields: ["data.differences_counts.fewer_ballots_count"],
+      }),
+    ).toBe(false);
+
+    expect(
+      isGlobalValidationResult({
+        code: "W301",
+        fields: ["data.votes_counts.blank_votes_count"],
+      }),
+    ).toBe(false);
+  });
+
+  test("hasOnlyGlobalValidationResults", () => {
+    const onlyGlobalResults: ClientValidationResult[] = [
+      {
+        code: "F204",
+        fields: ["data.votes_counts.votes_candidates_counts", "data.political_group_votes"],
+      },
+    ];
+
+    expect(hasOnlyGlobalValidationResults(onlyGlobalResults)).toBe(true);
+
+    const onlyLocalResults: ClientValidationResult[] = [
+      {
+        code: "W201",
+        fields: ["data.votes_counts.blank_votes_count"],
+        isGlobal: false,
+      },
+      {
+        code: "W202",
+        fields: ["data.votes_counts.invalid_votes_count"],
+        isGlobal: false,
+      },
+    ];
+
+    expect(hasOnlyGlobalValidationResults(onlyLocalResults)).toBe(false);
+
+    const mixedResults: ClientValidationResult[] = [
+      {
+        code: "F204",
+        fields: ["data.votes_counts.votes_candidates_counts", "data.political_group_votes"],
+        isGlobal: true,
+      },
+      {
+        code: "W201",
+        fields: ["data.votes_counts.blank_votes_count"],
+        isGlobal: false,
+      },
+    ];
+
+    expect(hasOnlyGlobalValidationResults(mixedResults)).toBe(false);
+  });
+
+  test("getErrorsAndWarnings errors and clientWarnings", () => {
+    const errors: ValidationResult[] = [
+      {
+        code: "F201",
+        fields: [
+          "data.voters_counts.total_admitted_voters_count",
+          "data.voters_counts.poll_card_count",
+          "data.voters_counts.proxy_certificate_count",
+          "data.voters_counts.voter_card_count",
+        ],
+      },
+      {
+        code: "F202",
+        fields: [
+          "data.votes_counts.total_votes_cast_count",
+          "data.votes_counts.votes_candidates_counts",
+          "data.votes_counts.blank_votes_count",
+          "data.votes_counts.invalid_votes_count",
+        ],
+      },
+    ];
+
+    const warnings: ValidationResult[] = [
+      {
+        code: "W201",
+        fields: ["data.votes_counts.blank_votes_count"],
+      },
+    ];
+
+    const clientWarnings: FieldValidationResult[] = [
+      {
+        code: "W201",
+        id: "blank_votes_count",
+      },
+    ];
+
+    const errorsAndWarnings = getErrorsAndWarnings(errors, warnings, clientWarnings);
+    expect(errorsAndWarnings.get("blank_votes_count")).toBeDefined();
+    expect(errorsAndWarnings.get("blank_votes_count")?.errors.length).toBe(1);
+    //warnings should not be added if errors.
+    expect(errorsAndWarnings.get("blank_votes_count")?.warnings.length).toBe(1);
```

Comment:
The comment states that warnings should not be added if there are errors but the length of the warnings is 1, so they are added?

Reply:
The warning that's present is from the `clientWarnings` defined in the test. We're missing a test with errors, warnings, and no client warnings.

Or we should validate on something more specific than the length of an array, which might be a good idea in any case.

Reply:
I made the test more specific.
Also the clientWarnings will hopefully be removed when moving logic to Input component.